syntax = "proto3";

package stt;

service STTService {
  rpc StreamAudio (stream AudioChunk) returns (stream Transcript);
}

message AudioChunk {
  string session_id = 1;
  bytes audio = 2;        
  int32 sample_rate = 3; 
}

message Transcript {
  string session_id = 1;
  string text = 2;
  bool is_final = 3;
  string language = 4;
}
